modules/social_features/social_group/modules/social_group_request/src/Form/GroupRequestMembershipRequestAnonymousForm.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Drupal\social_group_request\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\Core\Url;
+use Drupal\group\Entity\GroupInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Provides a form to request group membership for anonymous.
+ */
+class GroupRequestMembershipRequestAnonymousForm extends FormBase {
```

Comment:
It should be a controller because it doesn't contain any form elements as input field, button, etc.